pkgs/development/tools/phantomjs2/default.nix

```diff
@@ -104,6 +104,10 @@ in stdenv.mkDerivation rec {
     rm -r ../__nix_qt5__
   '';
 
+  postFixup = ''
```

Comment:
The default `QT_QPA_PLATFORM_PLUGIN_PATH` is set automatically by the library. Wrapping Qt applications breaks desktop integration. The "missing plugin" problem is solved by installing the package into an environment.

Reply:
@cocreature No, unfortunately. I don't know how to support `nix-shell` and desktop integration for Qt applications at the same time. If we set paths in a wrapper, then `nix-shell` works, but the applications break under a desktop environment. If we don't set paths in the wrapper, the opposite.

Reply:
@Mic92 `QT_QPA_PLATFORM_PLUGIN_PATH` doesn't get any value. Our Qt is patched to search in the system and user environments for plugins (platform or otherwise). Installation simple ensures that the required plugins are installed in one of those places.

I think it should be OK for you to set that variable in `shellHook`.